demo.py

```diff
@@ -23,52 +24,91 @@
     'data_format', 'NCHW', 'Data format: NCHW (gpu only) / NHWC')
 tf.app.flags.DEFINE_string(
     'ckpt_file', './saved_model/model.ckpt', 'Checkpoint file')
+tf.app.flags.DEFINE_string(
+    'frozen_model', '', 'Frozen tensorflow protobuf model')
 tf.app.flags.DEFINE_bool(
     'tiny', False, 'Use tiny version of YOLOv3')
 
-tf.app.flags.DEFINE_integer('size', 416, 'Image size')
+tf.app.flags.DEFINE_integer(
+    'size', 416, 'Image size')
 
-tf.app.flags.DEFINE_float('conf_threshold', 0.5, 'Confidence threshold')
-tf.app.flags.DEFINE_float('iou_threshold', 0.4, 'IoU threshold')
+tf.app.flags.DEFINE_float(
+    'conf_threshold', 0.5, 'Confidence threshold')
+tf.app.flags.DEFINE_float(
+    'iou_threshold', 0.4, 'IoU threshold')
+
+tf.app.flags.DEFINE_float(
+    'gpu_memory_fraction', 0.8, 'Gpu memory fraction to use')
 
 
 def main(argv=None):
-    if FLAGS.tiny:
-        model = yolo_v3_tiny.yolo_v3_tiny
-    else:
-        model = yolo_v3.yolo_v3
+
+    gpu_options = tf.GPUOptions(per_process_gpu_memory_fraction=FLAGS.gpu_memory_fraction)
 
     img = Image.open(FLAGS.input_img)
     img_resized = img.resize(size=(FLAGS.size, FLAGS.size))
 
     classes = load_coco_names(FLAGS.class_names)
 
-    # placeholder for detector inputs
-    inputs = tf.placeholder(tf.float32, [1, FLAGS.size, FLAGS.size, 3])
+    if FLAGS.frozen_model:
 
-    with tf.variable_scope('detector'):
-        detections = model(inputs, len(classes),
-                           data_format=FLAGS.data_format)
+        t0 = time.time()
```

Comment:
Maybe wrap this 2 if branches into functions which returns inputs and boxes?

Reply:
Made a function to wrap this, but I'm not so certain about how I handled it. Your input would be welcomed.

Reply:
I think I was severely overthinking this. I've simplified things a bit. Let me know if it is what you had in mind.